docs/installing-solidity.rst

```diff
@@ -32,102 +32,119 @@ Details about the usage of the Node.js package can be found in the
+.. _CMake: https://cmake.org/download/
+.. _Visual Studio 2015: https://www.visualstudio.com/products/vs-2015-product-editions
 
-    sudo add-apt-repository -y ppa:ethereum/ethereum
-    sudo add-apt-repository -y ppa:ethereum/ethereum-dev
-    sudo apt-get -y update
-    sudo apt-get -y upgrade # this will update cmake to version 3.x
-    sudo apt-get -y install libcryptopp-dev libjsoncpp-dev
 
-Ubuntu Xenial (16.04)
+External Dependencies
 ---------------------
 
-Below are the instructions to install the minimal dependencies required
-to compile Solidity on Ubuntu 16.04 (Xenial Xerus).
+We now have a "one button" script which installs all required external dependencies
+on macOS, Windows and on numerous Linux distros.  This used to be a multi-step
+manual process, but is now a one-liner:
 
-One of the dependencies (Crypto++ Library, with version >= 5.6.2) can be
-installed either by adding the Ethereum PPA (Option 1) or by backporting
-``libcrypto++`` from Ubuntu Development to Ubuntu Xenial (Option 2).
+.. code:: bash
 
-.. code-block:: bash
+    ./scripts/install_deps.sh
 
-    sudo apt-get -y install build-essential git cmake libgmp-dev libboost-all-dev \
-        libjsoncpp-dev
+Or, on Windows:
 
-    # (Option 1) For those willing to add the Ethereum PPA:
-    sudo add-apt-repository -y ppa:ethereum/ethereum
-    sudo add-apt-repository -y ppa:ethereum/ethereum-dev
-    sudo apt-get -y update
-    sudo apt-get -y upgrade
-    sudo apt-get -y install libcryptopp-dev
+.. code:: bash
 
-    ## (Option 2) For those willing to backport libcrypto++:
-    #sudo apt-get -y install ubuntu-dev-tools
-    #sudo pbuilder create
-    #mkdir ubuntu
-    #cd ubuntu
-    #backportpackage --workdir=. --build --dont-sign libcrypto++
-    #sudo dpkg -i buildresult/libcrypto++6_*.deb buildresult/libcrypto++-dev_*.deb
-    #cd ..
+    scripts\install_deps.bat
 
-Building
---------
 
-Run this if you plan on installing Solidity only:
+Command-Line Build
+------------------
 
-.. code-block:: bash
+Building Solidity is quite similar on Linux, macOS and other Unices:
+
+.. code:: bash
 
-    git clone --recursive https://github.com/ethereum/solidity.git
-    cd solidity
     mkdir build
     cd build
     cmake .. && make
 
-If you want to help developing Solidity,
-you should fork Solidity and add your personal fork as a second remote:
+And even on Windows:
 
-.. code-block:: bash
+.. code:: bash
 
-    cd solidity
-    git remote add personal git@github.com:username/solidity.git
+    mkdir build
+    cd build
+    cmake -G "Visual Studio 14 2015 Win64" ..
+
+This latter set of instructions should result in the creation of
+**solidity.sln** in that build directory.  Double-clicking on that file
+should result in Visual Studio firing up.  We suggest building
+**RelWithDebugInfo** configuration, but all others work.
+
+Alternatively, you can build for Windows on the command-line, like so:
+
+.. code:: bash
+
+    cmake --build . --config RelWithDebInfo
```

Comment:
We could add a last paragraph explaining where the binary will appear and perhaps even telling users to do `sudo make install`.